### PR TITLE
Add a command to allow removal of existing annotations

### PR DESCRIPTION
**Pull request by ticky**

### CODE REVIEW

**Comment by ticky** [comment]
This… somehow breaks the BSD build? 😔

**Comment by sj26** [comment]
It feels odd to me to add this to the existing command. Could we perhaps add a new command?

There's good precedence for `buildlite-agent noun verb`. Perhaps `buildkite-agent annotation remove --context=blah`                                                                                     would be a good choice?

File: clicommand/annotation_remove.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    cli.StringFlag{
+      Name:   "context",
+      Value:  "default",
+      Usage:  "The context of the annotation used to differentiate this annotation from others",
+      EnvVar: "BUILDKITE_ANNOTATION_CONTEXT",
+    },
+    cli.StringFlag{
+      Name:   "job",
+      Value:  "",
+      Usage:  "Which job should the annotation come from",
```
Comment (sj26) [inline clicommand/annotation_remove.go:59]:
```suggestion
      Usage:  "Which job is removing the annotation",
```

**Review by sj26 — DISMISSED** [review]
Great! I gave this a go against the corresponding agent API changes and it works a treat. :+1:

File: clicommand/annotation_remove.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+var AnnotationRemoveHelpDescription = `Usage:
+
+   buildkite-agent annotation remove [arguments...]
+
+Description:
+
+   Remove an existing annotation which was previously published using the
+   buildkite-agent annotate command.
+
+   Or if you leave context blank, it will use the default context.
```
Comment (ticky) [inline clicommand/annotation_remove.go:21]:
```suggestion
   If you leave context blank, it will use the default context.
```

**Comment by ticky** [comment]
Using my privileges to merge this because the code changes since the review were not meaningful.

**Pull request merged**